plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -604,6 +605,17 @@ private void invalidateTableStatisticsCache(String databaseName, String tableNam
                 .forEach(tableCache::invalidate);
     }
 
+    private void invalidateTablesWithParameterCache(String databaseName, String tableName)
+    {
+        tablesWithParameterCache.asMap().keySet().stream()
+                .filter(cacheKey -> cacheKey.getDatabaseName().equals(databaseName))
+                .filter(cacheKey -> {
+                    List<String> cacheValue = tablesWithParameterCache.getIfPresent(cacheKey);
```

Comment:
Shouldn't we implement `entrySet()` in `EvictableCache` and use that to avoid looking up each item?

Reply:
@electrum see https://github.com/trinodb/trino/pull/10845#discussion_r821058122